### ISSUE TITLE
PlayerShouldTakeDamage not called clientside

### ISSUE THREAD

**Issue by markusmarkusz** [Facepunch/garrysmod-issues#3020]
### Details
`GM:PlayerShouldTakeDamage()` isn't called shared.

### Steps to reproduce
Add a hook to test it.

As far as I know, it was a shared hook. (Somewhere I read it was a predicted hook.)

**Comment by daunknownman2010** [comment]
http://wiki.garrysmod.com/page/GM/PlayerShouldTakeDamage
It's called server-side only.

**Comment by Kefta** [comment]
http://wiki.garrysmod.com/index.php?title=GM/PlayerShouldTakeDamage&diff=16933&oldid=12999 It's still in shared.lua of the base gamemode.

**Comment by markusmarkusz** [comment]
Somewhere (I think in the Sandbox gamemode.) there is a comment that says PlayerShouldTakeDamage is shared.

**Comment by Kefta** [comment]
Yup https://github.com/garrynewman/garrysmod/blob/784cd57576d85712fa13a7cea3a9523b4df966b0/garrysmod/gamemodes/sandbox/gamemode/init.lua#L80-L86

**Comment by robotboy655** [comment]
It technically is a shared hook as it can be called from the engine to Lua on client, but there's nothing in the engine that triggers the conditions on the client which is why it is never called on client in Lua.

**Comment by markusmarkusz** [comment]
The comment in Sandbox says:
```
This is a shared function - the client will think they can 
	 damage the players even though they can't. This just means the 
	 prediction will show blood.
```
If I understood correctly, this hook was responsible for seeing blood if you shoot at someone, right? Which hook does this now? Is there any reason left to have this hook shared?

**Comment by Bo98** [comment]
ScalePlayerDamage should work for stopping blood particle effects.
